thanks/thanks.py

```diff
@@ -1,82 +1,97 @@
+            next(requirements.parse(r))
+            for r in requirements_list
+        ]
+        for req in reqs:
+            self.update_give_thanks(req.name)
+        self._display_thanks()
+
+    def _display_thanks(self):
+        if self.give_thanks_to:
+            cprint(
+                'You depend on {} who would {}'.format(
+                    colored('{} authors'.format(len(self.give_thanks_to)), 'cyan'),
+                    colored('enjoy donations!', 'green'),
+                ), attrs=['bold'])
+            colorized_data = [
+                ProjectData(
+                    name=colored(pd.name, 'cyan'),
+                    funding_link=colored(pd.funding_link, 'green'),
+                    authors=colored(pd.authors, 'yellow'),
+                )
+                for pd in self.give_thanks_to.values()
+            ]
+
+            print(format_pretty_table(
+                data=colorized_data,
+                column_names=['Project', 'Where to thank', 'Authors'],
+                horizontal_bar=' ',
+                vertical_bar=' ',
+            ))
+
+    def get_local_data(self, project_name):
+        if project_name in self.data:
+            data = ProjectData(
+                name=project_name,
+                funding_link=self.data[project_name]['url'],
+                authors=self.data[project_name]['authors']
+            )
+        else:
+            data = None
+        return data
+
+    def get_remote_data(self, project_name):
+        url_format = 'https://pypi.org/pypi/{}/json'.format
+        try:
+            resp = requests.get(url_format(project_name))
+            project_data = resp.json()
+            data = ProjectData(
+                name=project_name,
+                funding_link=project_data['info'].get('funding_url', '*** No funding url available ***'),
+                authors=project_data['info'].get('author', '')
+            )
+        except requests.exceptions.ConnectionError:
+            data = None
+        except json.decoder.JSONDecodeError:
+            data = None
+        return data
 
     def update_give_thanks(self, package_name):
         if self.debug:
             print('Checking ', package_name)
-        try:
-            roles = client.package_roles(package_name)
-            role_names = set(role[1] for role in roles)
-            if self.debug:
-                print('Checking role names: ', role_names)
-            for name in role_names:
-                if name in self.give_thanks_to:
-                    self.give_thanks_to[name]['packages'].append(package_name)
-                else:
-                    if name in self.data:
-                        self.give_thanks_to[name] = {
-                            'url': self.data[name],
-                            'packages': [package_name],
-                        }
-        except TypeError:
-            pass
+        package_data = self.get_local_data(package_name) or self.get_remote_data(package_name)
```

Comment:
I think I want to default to local data unless the user passes a flag like `fetch-remote` or something... or if they give us a single package name we could go fetch that? I'd love for `thanks` to work without an outbound internet connection if possible.

Reply:
So as it stands a remote fetch only occurs if there is no local data.
Currently this is from our mock db. But this will soon be taken from the metadata of installed packages.

Does this follow your intention? Or were you suggesting that we only do a remote fetch with an explicit flag? I'd be more inclined to have a flag with disabled remote fetching (or read it from setup.cfg or other config)